docs/pages/api-reference/liveblocks-client.mdx

```diff
@@ -891,18 +891,142 @@ Note that the function should be called like this `defineAiTool()({ ... })`
+
+1. Implement `execute` _and_ `render`
+1. Implement only `execute`, but no `render`
+1. Implement only `render`, but make sure to eventually call `respond()`
+
+#### Implementing your tool call via `execute` [#implement-via-execute]
+
+If you implement the `execute` function, this function will automatically be
+invoked when the tool call gets made. The return value of this function will be
+the result that will be passed back to the AI copilot.
+
+- `{ data: any, description?: string }` The data to return in case of success.
+  `data` must be a legal JSON value. Providing a description is optional. If you
+  provide a description, it will be passed to the AI copilot to help it
+  understand the returned data or to provide follow-up instructions for how to
+  respond to this tool result.
+- `{ error: string }` The error message in case the tool call failed to execute.
+- `{ cancel: true | string }` If the tool call should be cancelled. You can
+  optionally provide a cancel reason as an instruction to the AI copilot.
+
+The returned value can be observed in the `render` method, through the `result`
+param:
+
+```tsx
+defineAiTool()({
+  /* ... */
+
+  execute: async () => {
+    await sleep(1000);
+    return { data: { user: { name: "Alice" } } };
+  },
+
+  render: ({ result }) => {
+    if (result.data) {
+      return <div>Found user: {result.data.user.name}</div>;
+    }
+
+    // Tool hasn't executed yet
+    return <Spinner />;
+  },
+});
+```
+
+If you do not implement `render` alongside `execute`, the tool call will still
+be executed, but no UI will be displayed. The result will still be passed back
+to the AI copilot.
+
+#### Implementing your tool call via `render` [#implement-via-render]
+
+Sometimes you may not want to immediately execute the tool call. This is most
+common to build a Human-in-the-Loop (HITL) style UI where you want the user to
+confirm or correct the tool call’s behavior. In these scenarios, you do not want
+to implement `execute`. Instead, you could display any UI, as long as you
+eventually call the `respond` function that is provided to `render`’s props.
+
+```tsx
+defineAiTool()({
+  /* ... */
+  /* NOTE: No execute method used here! */
+
+  render: ({ respond }) => {
+    return (
+      <div>
+        <button
+          onClick={() => {
+            respond({ data: { user: { name: "Alice" } } });
+          }}
+        >
+          Confirm
+        </button>
+        <button
+          onClick={() => {
+            respond({ cancel: true });
+          }}
+        >
+          Confirm
+        </button>
+      </div>
+    );
+  },
+});
+```
+
+In this example, until the Confirm button is clicked, the AI chat will remain in
+“executing” stage, awaiting the result of this tool call.
+
+This example is for illustrative purposes only. In practice, using our
+[`AiTool.Confirmation`](/docs/api-reference/liveblocks-react-ui#AiTool.Confirmation)
+tool is preferred.
```

Comment:
Only for simple yes/no confirmations, other valid use-cases like questions/forms aren't covered by it.